### PR TITLE
Add Expense.quote resolver and Wise quote caching

### PR DESCRIPTION
**Pull request by kewitz**
Solves https://github.com/opencollective/opencollective/issues/3089.
Solves https://github.com/opencollective/opencollective/issues/4963.

This allows us to display Wise fees before paying for the expense and optimizing the performance of Wise expense payment flow.

### CODE REVIEW

File: server/paymentProviders/transferwise/index.ts [hunk 1]
```diff
@@ -104,6 +104,15 @@ async function quoteExpense(
 ): Promise<QuoteV2> {
   await populateProfileId(connectedAccount);
 
+  const isExistingQuoteValid =
+    expense.data?.quote &&
+    expense.data.quote.paymentOption &&
+    moment.utc().subtract(60, 'seconds').isBefore(expense.data.quote.expirationTime);
```
Comment (znarf) [inline server/paymentProviders/transferwise/index.ts:110]: What's the rationale for those 60 seconds?

What is the expirationTime in practice?

Comment (kewitz) [reply server/paymentProviders/transferwise/index.ts:110]: I'm considering a 60 seconds buffer for the transaction creation and fund to be complete. Basically, the whole process that is sometimes taking 25 seconds and causing the network error in https://github.com/opencollective/opencollective/issues/4963.

The expiration time is related to how long does the Quote you created to fulfill the transaction will be valid.

Comment (znarf) [reply server/paymentProviders/transferwise/index.ts:110]: What is a normal value for `expirationTime` in practice?

**Review by znarf — APPROVED** [review]
Looks good, please answer the question on expirationTime so I understand better.

File: server/paymentProviders/transferwise/index.ts (hunk 1 above)
Comment (kewitz) [reply server/paymentProviders/transferwise/index.ts:110]: ISO date string.

Comment (znarf) [reply server/paymentProviders/transferwise/index.ts:110]: Rephrasing, I just wanted to ask how far it is in the future.

**Review by Claudio060320 — COMMENTED** [review]
Duplicate of #

**Pull request merged**